mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -82,6 +86,31 @@ public interface ResourceCluster extends ResourceClusterGateway {
      */
     CompletableFuture<Ack> refreshClusterScalerRuleSet();
 
+    /**
+     * Disables task executors that match the passed set of attributes
+     *
+     * @param attributes attributes that need to be present in the task executor's set of attributes.
+     * @param expiry     instant at which the request can be marked as complete. this is important because we cannot be constantly checking if new task executors match the disabled criteria or not.
+     * @return a future that completes when the underlying operation is registered by the system
+     */
+    CompletableFuture<Void> disableTaskExecutorsFor(Map<String, String> attributes, Instant expiry);
```

Comment:
nit: will CompletableFuture<Ack> be more consistent?

Reply:
+1